app/services/optout/OptOutService.scala

```diff
@@ -146,16 +148,31 @@ class OptOutService @Inject()(itsaStatusUpdateConnector: ITSAStatusUpdateConnect
     setupOptOutProposition().map { proposition =>
       proposition.optOutPropositionType.flatMap {
         case p: OneYearOptOutProposition => Some(ConfirmedOptOutViewModel(optOutTaxYear = p.intent, state = p.state()))
-        case p: MultiYearOptOutProposition => intent.map(i => ConfirmedOptOutViewModel(optOutTaxYear = i, state = p.state()))
+        case p: MultiYearOptOutProposition => intent.map(i => ConfirmedOptOutViewModel(optOutTaxYear = i, state = p.state()))//todo: add test code
       }
     }
   }
 
+
+  def getTaxYearsAvailableForOptOut()(implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Future[Seq[TaxYear]] = {
+    setupOptOutProposition().map(proposition => proposition.availableTaxYearsForOptOut)
+  }
+
+  def getSubmissionCountForTaxYear(offeredOptOutYears: Seq[TaxYear])
+                                  (implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Future[Map[Int, Int]] = {
+    val futureCounts: Seq[Future[(Int, Int)]] = offeredOptOutYears.map { optOutTaxYear =>
+      nextUpdatesService.getSubmissionCounts(optOutTaxYear)
+    }
+
+    Future.sequence(futureCounts).map(_.toMap)
+  }
+
 }
 
 
 object OptOutService {
-  def combineByReturningAnyFailureFirstOrAnySuccess(responses: Seq[Future[OptOutUpdateResponse]])(implicit ec: ExecutionContext): Future[OptOutUpdateResponse] = {
+
+  private def combineByReturningAnyFailureFirstOrAnySuccess(responses: Seq[Future[OptOutUpdateResponse]])(implicit ec: ExecutionContext): Future[OptOutUpdateResponse] = {
```

Comment:
could fix:  rename the function, like for eg.`aggregateResponses` or `combineResults`

Reply:
I'm happy with the name as it is, it's documenting the logic and saving me the time to review and understand the logic which isn't trivial if you're trying to quickly understand what it's doing

Reply:
done